bibliopixel/drivers/SimPixel/__init__.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python
+# encoding: utf-8
+from .. driver_base import DriverBase, ChannelOrder
+
+import logging
+import signal
+import sys
+import time
+import threading
+import uuid
+import struct
+from ... import log
+
+from . SimpleWebSocketServer import WebSocket, SimpleWebSocketServer
+
+
+class SimPixelWebSocket(WebSocket):
+
+    def __init__(self, *args, driver, layout):
+        super(SimPixelWebSocket, self).__init__(*args)
+        self.driver = driver
+        self.connected = False
+        self.layout = layout
+        self.oid = None
+        log.debug('Server started...')
+
+    def handleConnected(self):
+        log.debug('Connected:{}'.format(self.address))
+        self.connected = True
+        # self.kinect = KinectFactory.create_kinect()
+        self.oid = uuid.uuid1()
+        self.driver.add_websock(self.oid, self.send_pixels)
+        self.sendMessage(bytearray([0x00, 0x00]) + self.layout)
+
+    def handleClose(self):
+        self.driver.remove_websock(self.oid)
+        self.connected = False
+        log.debug('Closed:{}'.format(self.address))
+
+    def handleMessage(self):
+        # degs = int(float(self.data))
+        # self.kinect.set_tilt(degs)
+        pass
+
+    def send_pixels(self, pixels):
+        if self.connected:
+            self.sendMessage(bytearray([0x00, 0x01]) + pixels)
+            # print(time.time())
+
+
+class ws_thread(threading.Thread):
+    def __init__(self, server):
+        super(ws_thread, self).__init__()
+        self.server = server
+
+    def run(self):
+        try:
+            self.server.serveforever()
+        except:
+            pass
+        log.debug('WebSocket Server closed')
+
+
+class DriverSimPixel(DriverBase):
+
+    def __init__(self, num, port=1337, layout=None):
+        super(DriverSimPixel, self).__init__(num)
+        self.port = port
+        self.layout = None
+        self.server = self.ws_thread = None
+        self.websocks = {}
+
+        if layout:
+            self.set_layout(layout)
+
+    def __start_server(self):
+        log.debug('Starting server...')
+        self.server = SimpleWebSocketServer('', self.port, SimPixelWebSocket,
+                                            driver=self, layout=self.layout,
+                                            selectInterval=0.001)
+        self.ws_thread = ws_thread(self.server)
```

Comment:
I'm perfectly happy with that... when I started that I think I thought it would need other functionality like some of the other thread's I've done, but you are right, this is simple and could be changed.